### PR TITLE
operator: remove the generated ssh key machineconfig

### PR DESCRIPTION
**Pull request by abhinavdahiya**
The original feature of syncing the ssh key for authorized users from `cluster-config-v1` config map was added in [1]. The original assumption was
that as `cluster-config-v1` was being deprecated in favor of global configs [2] and we will be able to find new home for the source of authorized keys for `core` user.

Based on several discussions like [3], The better UX for users is that machineconfig with authorized keys is not owned by the controller and is rather pushed by the installer at install time.
This allows users to edit the machineconfig directly to remove, add and manage the list of authorized keys.

This requires https://github.com/openshift/installer/pull/1150

/cc @kikisdeliveryservice @crawford @cgwalters

### CODE REVIEW

**Comment by ashcrow** [comment]
```
level=fatal msg="waiting for openshift-console URL: context deadline exceeded"
```
/test e2e-aws-op

**Comment by runcom** [comment]
needs to be rebased as per our conversation on slack - we'll get this in to avoid a migration post release

**Comment by runcom** [comment]
/hold cancel

we're waiting on #548 as well before this anyway

**Comment by runcom** [comment]
alrighty, this is good to be rebased reviewed and merged

**Comment by kikisdeliveryservice** [comment]
i do think #541 should go in before this

**Comment by runcom** [comment]
#541 went it, @abhinavdahiya can you rebase this so we can review and merge?

**Comment by abhinavdahiya** [comment]
> #541 went it, @abhinavdahiya can you rebase this so we can review and merge?

@runcom PTAL

**Comment by runcom** [comment]
so nice to see this (cc @kikisdeliveryservice )

```
=== RUN   TestUpdateSSH
--- PASS: TestUpdateSSH (230.63s)
```
/approve

@kikisdeliveryservice @cgwalters ptal for merging

**Comment by cgwalters** [comment]
LGTM

File: docs/Update-SSHKeys.md [hunk 1]
```diff
@@ -42,85 +58,100 @@ spec:
...
-    01-worker-container-runtime                                 4.0.0-alpha.0-83-g0a84145f-dirty   2.2.0             87m
-    01-worker-kubelet                                           4.0.0-alpha.0-83-g0a84145f-dirty   2.2.0             87m
-    99-master-9c65f9fb-41d0-11e9-994d-02360d172130-registries   4.0.0-alpha.0-83-g0a84145f-dirty   2.2.0             84m
-    99-worker-9c67b221-41d0-11e9-994d-02360d172130-registries   4.0.0-alpha.0-83-g0a84145f-dirty   2.2.0             84m
-    rendered-master-a1884339a91f02f19898fe9c5929256b            4.0.0-alpha.0-83-g0a84145f-dirty   2.2.0             87m
-    rendered-worker-777a036887bb25188f47c6d3219eabb9            4.0.0-alpha.0-83-g0a84145f-dirty   2.2.0             2s
-    rendered-worker-e99b9948bccfd387c71397b31e2c985e            4.0.0-alpha.0-83-g0a84145f-dirty   2.2.0             87m
-    ssh-workers-01                                                                                                   7s
+You should see the new MachineConfig name appear immediately, from our example config:
+
```
Comment (kikisdeliveryservice) [inline docs/Update-SSHKeys.md]: This output is out of date, update it so it is correct, see PR #533

**Comment by cgwalters** [comment]
> or will the 99-worker-ssh NOT be updated 

AIUI it's more like "99-worker-ssh is created by the installer if a SSH key was provided to it" - after that it's user editable (including deletion).  There's no "internal controller" for it like there are for the `00-*` configs etc.

The user could create separately named SSH MCs too - the keys would then be unioned.

File: docs/Update-SSHKeys.md [hunk 2]
```diff
@@ -42,85 +58,100 @@ spec:
...
-    01-master-kubelet                                           4.0.0-alpha.0-83-g0a84145f-dirty   2.2.0             87m
-    01-worker-container-runtime                                 4.0.0-alpha.0-83-g0a84145f-dirty   2.2.0             87m
-    01-worker-kubelet                                           4.0.0-alpha.0-83-g0a84145f-dirty   2.2.0             87m
-    99-master-9c65f9fb-41d0-11e9-994d-02360d172130-registries   4.0.0-alpha.0-83-g0a84145f-dirty   2.2.0             84m
-    99-worker-9c67b221-41d0-11e9-994d-02360d172130-registries   4.0.0-alpha.0-83-g0a84145f-dirty   2.2.0             84m
-    rendered-master-a1884339a91f02f19898fe9c5929256b            4.0.0-alpha.0-83-g0a84145f-dirty   2.2.0             87m
-    rendered-worker-777a036887bb25188f47c6d3219eabb9            4.0.0-alpha.0-83-g0a84145f-dirty   2.2.0             2s
-    rendered-worker-e99b9948bccfd387c71397b31e2c985e            4.0.0-alpha.0-83-g0a84145f-dirty   2.2.0             87m
-    ssh-workers-01                                                                                                   7s
+You should see the new MachineConfig name appear immediately, from our example config:
```
Comment (kikisdeliveryservice) [inline docs/Update-SSHKeys.md]: does this sentence make sense anymore?

File: docs/Update-SSHKeys.md (hunk 1 above)
Comment (abhinavdahiya) [reply docs/Update-SSHKeys.md]: updated.

File: docs/Update-SSHKeys.md (hunk 2 above)
Comment (abhinavdahiya) [reply docs/Update-SSHKeys.md]: updated

File: pkg/controller/template/render.go [hunk 3]
```diff
@@ -95,14 +95,7 @@ func GenerateMachineConfigsForRole(config *RenderConfig, role string, path strin
 		return nil, fmt.Errorf("failed to read dir %q: %v", path, err)
 	}
 
-	sshMachineConfigForRole, err := generateSSHConfig(config.SSHKey, role)
```
Comment (kikisdeliveryservice) [inline pkg/controller/template/render.go]: @abhinavdahiya  l ast thing: if we are deleting this, then we dont need the func() generateSSHConfig  (below) anymore right? this is the only spot that calls that function.

Comment (abhinavdahiya) [reply pkg/controller/template/render.go]: done

Comment (kikisdeliveryservice) [reply pkg/controller/template/render.go]: thanks!!!

**Comment by kikisdeliveryservice** [comment]
It's time! :+1: 

/lgtm

**Comment by abhinavdahiya** [comment]
/hold
would need https://github.com/openshift/installer/pull/1150 to go in so that we are not left without SSH keys being setup.

**Comment by abhinavdahiya** [comment]
https://github.com/openshift/installer/pull/1150 merged 
/hold cancel

**Comment by abhinavdahiya** [comment]
Hmm... haven't seen that before :/
```
could not wait for build: the build machine-config-controller failed after 3m16s with reason PushImageToRegistryFailed: Failed to push the image to the registry.

Registry server Address: 
Registry server User Name: serviceaccount
Registry server Email: serviceaccount@example.org
Registry server Password: <<non-empty>>
error: build error: Failed to push image: Put https://dock...t canceled (Client.Timeout exceeded while awaiting headers)
```
/retest

**Comment by runcom** [comment]
looks like this may have broken upgrades (investigating with QE) - might have a clue related to  the "generated by controller version" and the check the MCO does on that field.

**Pull request merged**